site/profile/manifests/nodejs.pp

```diff
@@ -0,0 +1,16 @@
+class profile::nodejs (
+        String $version,
+  Array[String] $npm_packages,
+)
+{
+  class { 'nodejs':
+    repo_url_suffix => $version,
+  }
+  $npm_packages.each |String $npm_package| {
+    package { $npm_package:
+    ensure   => 'present',
+    provider => 'npm',
```

Comment:
Since I don't think npm is a provider shipped by default, remember to check if it's necessary to add the npm module to Puppetfile in the control-repo